File: redome-front/src/app/shared/vo/pedido.exame.doador.internacional.vo.ts
import { BaseEntidade } from '../base.entidade';
import { ConvertUtil } from '../util/convert.util';

export class PedidoExameDoadorInternacionalVo extends BaseEntidade {
 
    private _idBusca: number;
    private _idSolicitacao: number;
    private _idStatusSolicitacao: number;
    private _idTipoSolicitacao: number;

    private _idDoador: number;
    private _veioDoEmdis: number;
    private _idOrigem: string;
    private _tipoDoador: number;

    private _idPedidoExame: number;
    private _idExame: number;
    private _origem: string;

    private _exame: string;

    private _idTarefa: number;
    private _idProcesso: number;
    private _tipoTarefa: number;

    private _dataCriacao: Date;

    private _grid: string;

    /**
     * Getter idBusca
     * @return {number}
     */
    public get idBusca(): number {
        return this._idBusca;
    }

    /**
     * Getter idSolicitacao
     * @return {number}
     */
    public get idSolicitacao(): number {
        return this._idSolicitacao;
    }

    /**
     * Getter idStatusSolicitacao
     * @return {number}
     */
    public get idStatusSolicitacao(): number {
        return this._idStatusSolicitacao;
    }

    /**
     * Getter idDoador
     * @return {number}
     */
    public get idDoador(): number {
        return this._idDoador;
    }

    /**
     * Getter VeioDoEmdis
     * @return {number}
     */
    public get veioDoEmdis(): number {
        return this._veioDoEmdis;
    }

    /**
     * Getter idOrigem
     * @return {string}
     */
    public get idOrigem(): string {
        return this._idOrigem;
    }

    /**
     * Getter tipoDoador
     * @return {number}
     */
    public get tipoDoador(): number {
        return this._tipoDoador;
    }

    /**
     * Getter idPedidoExame
     * @return {number}
     */
    public get idPedidoExame(): number {
        return this._idPedidoExame;
    }

    /**
     * Getter idExame
     * @return {number}
     */
    public get idExame(): number {
        return this._idExame;
    }

    /**
     * Getter Origem
     * @return {string}
     */
    public get origem(): string {
        return this._origem;
    }

    /**
     * Getter exame
     * @return {string}
     */
    public get exame(): string {
        return this._exame;
    }

    /**
     * Getter idTarefa
     * @return {number}
     */
    public get idTarefa(): number {
        return this._idTarefa;
    }

    /**
     * Getter idProcesso
     * @return {number}
     */
    public get idProcesso(): number {
        return this._idProcesso;
    }

    /**
     * Getter tipoTarefa
     * @return {number}
     */
    public get tipoTarefa(): number {
        return this._tipoTarefa;
    }

    /**
     * Getter dataCriacao
     * @return {Date}
     */
    public get dataCriacao(): Date {
        return this._dataCriacao;
    }

    /**
     * Setter idBusca
     * @param {number} value
     */
    public set idBusca(value: number) {
        this._idBusca = value;
    }

    /**
     * Setter idSolicitacao
     * @param {number} value
     */
    public set idSolicitacao(value: number) {
        this._idSolicitacao = value;
    }

    /**
     * Setter idStatusSolicitacao
     * @param {number} value
     */
    public set idStatusSolicitacao(value: number) {
        this._idStatusSolicitacao = value;
    }

    /**
     * Setter idDoador
     * @param {number} value
     */
    public set idDoador(value: number) {
        this._idDoador = value;
    }

    /**
     * Setter VeioDoEmdis
     * @param {number} value
     */
    public set veioDoEmdis(value: number) {
        this._veioDoEmdis = value;
    }

    /**
     * Setter idOrigem
     * @param {string} value
     */
    public set idOrigem(value: string) {
        this._idOrigem = value;
    }

    /**
     * Setter tipoDoador
     * @param {number} value
     */
    public set tipoDoador(value: number) {
        this._tipoDoador = value;
    }

    /**
     * Setter idPedidoExame
     * @param {number} value
     */
    public set idPedidoExame(value: number) {
        this._idPedidoExame = value;
    }

    /**
     * Setter idExame
     * @param {number} value
     */
    public set idExame(value: number) {
        this._idExame = value;
    }

    /**
     * Setter Origem
     * @param {string} value
     */
    public set origem(value: string) {
        this._origem = value;
    }

    /**
     * Setter exame
     * @param {string} value
     */
    public set exame(value: string) {
        this._exame = value;
    }

    /**
     * Setter idTarefa
     * @param {number} value
     */
    public set idTarefa(value: number) {
        this._idTarefa = value;
    }

    /**
     * Setter idProcesso
     * @param {number} value
     */
    public set idProcesso(value: number) {
        this._idProcesso = value;
    }

    /**
     * Setter tipoTarefa
     * @param {number} value
     */
    public set tipoTarefa(value: number) {
        this._tipoTarefa = value;
    }

    /**
     * Setter dataCriacao
     * @param {Date} value
     */
    public set dataCriacao(value: Date) {
        this._dataCriacao = value;
    }

    /**
     * Getter idTipoSolicitacao
     * @return {number}
     */
    public get idTipoSolicitacao(): number {
        return this._idTipoSolicitacao;
    }

    /**
     * Setter idTipoSolicitacao
     * @param {number} value
     */
    public set idTipoSolicitacao(value: number) {
        this._idTipoSolicitacao = value;
    }

    /**
     * Getter grid
     * @return {string}
     */
	public get grid(): string {
		return this._grid;
	}

    /**
     * Setter grid
     * @param {string} value
     */
	public set grid(value: string) {
		this._grid = value;
	}


    public jsonToEntity(res: any): this {

        this.idBusca = ConvertUtil.parseJsonParaAtributos(res.idBusca, new Number());
        this.idSolicitacao = ConvertUtil.parseJsonParaAtributos(res.idSolicitacao, new Number());
        this.idStatusSolicitacao = ConvertUtil.parseJsonParaAtributos(res.idStatusSolicitacao, new Number());
        this.idTipoSolicitacao = ConvertUtil.parseJsonParaAtributos(res.idTipoSolicitacao, new Number());

        this.idDoador = ConvertUtil.parseJsonParaAtributos(res.idDoador, new Number());
        this.veioDoEmdis = ConvertUtil.parseJsonParaAtributos(res.veioDoEmdis, new Number());
        this.idOrigem = ConvertUtil.parseJsonParaAtributos(res.idOrigem, new String());
        this.tipoDoador = ConvertUtil.parseJsonParaAtributos(res.tipoDoador, new Number());

        this.idPedidoExame = ConvertUtil.parseJsonParaAtributos(res.idPedidoExame, new Number());
        this.idExame = ConvertUtil.parseJsonParaAtributos(res.idExame, new Number());
        this.origem = ConvertUtil.parseJsonParaAtributos(res.origem, new String());

        this.exame = ConvertUtil.parseJsonParaAtributos(res.exame, new String());

        this.idTarefa = ConvertUtil.parseJsonParaAtributos(res.idTarefa, new Number());
        this.idProcesso = ConvertUtil.parseJsonParaAtributos(res.idProcesso, new Number());
        this.tipoTarefa = ConvertUtil.parseJsonParaAtributos(res.tipoTarefa, new Number());

        this.dataCriacao = ConvertUtil.parseJsonParaAtributos(res.dataCriacao, new Date());
        this.grid = ConvertUtil.parseJsonParaAtributos(res.grid, new String());

        return this;
    }
}

File: redome/src/main/java/br/org/cancer/modred/vo/PedidoExameDoadorInternacionalVo.java
package br.org.cancer.modred.vo;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 
 * Classe utilizada para criação da consulta de doador - contato passivo.
 * 
 * @author evandro.gomes
 *
 */
public class PedidoExameDoadorInternacionalVo implements Serializable, Cloneable {

	private static final long serialVersionUID = 4860435637249185865L;

	private Long idBusca;
	private Long idSolicitacao;
	private Long idStatusSolicitacao;
	private Long idTipoSolicitacao;
	private Long idTipoExame;
	private String nomeTipoExame;

	private Long idDoador;
	private Long veioDoEmdis;
	private String idOrigem;
	private Long tipoDoador;

	private Long idPedidoExame;
	private Long idExame;
	private String origem;

	private String exame;

	private Long idTarefa;
	private Long idProcesso;
	private Long tipoTarefa;

	private LocalDateTime dataCriacao;

	private Long idTipoServico;
	private String descricaoServico;
	private Long idPagamento;

	private String grid;

	public PedidoExameDoadorInternacionalVo() {
	}

	/**
	 * @return the idBusca
	 */
	public Long getIdBusca() {
		return idBusca;
	}

	/**
	 * @param idBusca the idBusca to set
	 */
	public void setIdBusca(Long idBusca) {
		this.idBusca = idBusca;
	}

	/**
	 * @return the idSolicitacao
	 */
	public Long getIdSolicitacao() {
		return idSolicitacao;
	}

	/**
	 * @param idSolicitacao the idSolicitacao to set
	 */
	public void setIdSolicitacao(Long idSolicitacao) {
		this.idSolicitacao = idSolicitacao;
	}

	/**
	 * @return the idStatusSolicitacao
	 */
	public Long getIdStatusSolicitacao() {
		return idStatusSolicitacao;
	}

	/**
	 * @param idStatusSolicitacao the idStatusSolicitacao to set
	 */
	public void setIdStatusSolicitacao(Long idStatusSolicitacao) {
		this.idStatusSolicitacao = idStatusSolicitacao;
	}

	/**
	 * @return the idDoador
	 */
	public Long getIdDoador() {
		return idDoador;
	}

	/**
	 * @param idDoador the idDoador to set
	 */
	public void setIdDoador(Long idDoador) {
		this.idDoador = idDoador;
	}

	/**
	 * @return the veioDoEmdis
	 */
	public Long getVeioDoEmdis() {
		return veioDoEmdis;
	}

	/**
	 * @param veioDoEmdis the veioDoEmdis to set
	 */
	public void setVeioDoEmdis(Long veioDoEmdis) {
		this.veioDoEmdis = veioDoEmdis;
	}

	/**
	 * @return the idOrigem
	 */
	public String getIdOrigem() {
		return idOrigem;
	}

	/**
	 * @param idOrigem the idOrigem to set
	 */
	public void setIdOrigem(String idOrigem) {
		this.idOrigem = idOrigem;
	}

	/**
	 * @return the tipoDoador
	 */
	public Long getTipoDoador() {
		return tipoDoador;
	}

	/**
	 * @param tipoDoador the tipoDoador to set
	 */
	public void setTipoDoador(Long tipoDoador) {
		this.tipoDoador = tipoDoador;
	}

	/**
	 * @return the idPedidoExame
	 */
	public Long getIdPedidoExame() {
		return idPedidoExame;
	}

	/**
	 * @param idPedidoExame the idPedidoExame to set
	 */
	public void setIdPedidoExame(Long idPedidoExame) {
		this.idPedidoExame = idPedidoExame;
	}

	/**
	 * @return the idExame
	 */
	public Long getIdExame() {
		return idExame;
	}

	/**
	 * @param idExame the idExame to set
	 */
	public void setIdExame(Long idExame) {
		this.idExame = idExame;
	}

	/**
	 * @return the origem
	 */
	public String getOrigem() {
		return origem;
	}

	/**
	 * @param origem the origem to set
	 */
	public void setOrigem(String origem) {
		this.origem = origem;
	}

	/**
	 * @return the exame
	 */
	public String getExame() {
		return exame;
	}

	/**
	 * @param exame the exame to set
	 */
	public void setExame(String exame) {
		this.exame = exame;
	}

	/**
	 * @return the idTarefa
	 */
	public Long getIdTarefa() {
		return idTarefa;
	}

	/**
	 * @param idTarefa the idTarefa to set
	 */
	public void setIdTarefa(Long idTarefa) {
		this.idTarefa = idTarefa;
	}

	/**
	 * @return the idProcesso
	 */
	public Long getIdProcesso() {
		return idProcesso;
	}

	/**
	 * @param idProcesso the idProcesso to set
	 */
	public void setIdProcesso(Long idProcesso) {
		this.idProcesso = idProcesso;
	}

	/**
	 * @return the tipoTarefa
	 */
	public Long getTipoTarefa() {
		return tipoTarefa;
	}

	/**
	 * @param tipoTarefa the tipoTarefa to set
	 */
	public void setTipoTarefa(Long tipoTarefa) {
		this.tipoTarefa = tipoTarefa;
	}

	/**
	 * @return the dataCriacao
	 */
	public LocalDateTime getDataCriacao() {
		return dataCriacao;
	}

	/**
	 * @param dataCriacao the dataCriacao to set
	 */
	public void setDataCriacao(LocalDateTime dataCriacao) {
		this.dataCriacao = dataCriacao;
	}

	/**
	 * @return the idTipoSolicitacao
	 */
	public Long getIdTipoSolicitacao() {
		return idTipoSolicitacao;
	}

	/**
	 * @param idTipoSolicitacao the idTipoSolicitacao to set
	 */
	public void setIdTipoSolicitacao(Long idTipoSolicitacao) {
		this.idTipoSolicitacao = idTipoSolicitacao;
	}

	/**
	 * @return the idTipoServico
	 */
	public Long getIdTipoServico() {
		return idTipoServico;
	}

	/**
	 * @param idTipoServico the idTipoServico to set
	 */
	public void setIdTipoServico(Long idTipoServico) {
		this.idTipoServico = idTipoServico;
	}

	/**
	 * @return the descricaoServico
	 */
	public String getDescricaoServico() {
		return descricaoServico;
	}

	/**
	 * @param descricaoServico the descricaoServico to set
	 */
	public void setDescricaoServico(String descricaoServico) {
		this.descricaoServico = descricaoServico;
	}

	/**
	 * @return the idPagamento
	 */
	public Long getIdPagamento() {
		return idPagamento;
	}

	/**
	 * @param idPagamento the idPagamento to set
	 */
	public void setIdPagamento(Long idPagamento) {
		this.idPagamento = idPagamento;
	}

	/**
	 * @return the grid
	 */
	public String getGrid() {
		return grid;
	}

	/**
	 * @param grid the grid to set
	 */
	public void setGrid(String grid) {
		this.grid = grid;
	}

	/**
	 * @return the idTipoExame
	 */
	public Long getIdTipoExame() {
		return idTipoExame;
	}

	/**
	 * @param idTipoExame the idTipoExame to set
	 */
	public void setIdTipoExame(Long idTipoExame) {
		this.idTipoExame = idTipoExame;
	}

	/**
	 * @return the nomeTipoExame
	 */
	public String getNomeTipoExame() {
		return nomeTipoExame;
	}

	/**
	 * @param nomeTipoExame the nomeTipoExame to set
	 */
	public void setNomeTipoExame(String nomeTipoExame) {
		this.nomeTipoExame = nomeTipoExame;
	}

	@Override
	public PedidoExameDoadorInternacionalVo clone() {
		PedidoExameDoadorInternacionalVo clone = new PedidoExameDoadorInternacionalVo();
		clone.setIdBusca(this.idBusca);
		clone.setIdSolicitacao(this.idSolicitacao);
		clone.setIdStatusSolicitacao(this.idStatusSolicitacao);
		clone.setIdTipoSolicitacao(this.idTipoSolicitacao);
		clone.setIdDoador(this.idDoador);
		clone.setVeioDoEmdis(this.veioDoEmdis);
		clone.setIdOrigem(this.idOrigem);
		clone.setTipoDoador(this.tipoDoador);
		clone.setOrigem(this.origem);
		clone.setDataCriacao(this.dataCriacao);
		return clone;
	}

}
